### ISSUE TITLE
Svix does not work with Cloudflare Worker

### ISSUE THREAD

**Issue by BernardoSM** [svix/svix-webhooks#1331]
### Version
^1.24.0

### Platform
Darwin Bernardos-MacBook-Pro.local 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:12:58 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6000 arm64

### Description

I'm not able to use Svix with Cloudflare Workers.

You can reproduce it doing the following steps:

1. Start a cloudflare worker project: https://developers.cloudflare.com/workers/get-started/guide/
2. Try to do the first step from svix quickstart: https://docs.svix.com/quickstart#creating-a-consumer-application
3. You'll see the following error: `The 'credentials' field on 'RequestInitializerDict' is not implemented`

I think the error is related to this issue: https://github.com/cloudflare/workers-sdk/issues/2514

**Comment by giautm** [comment]
The Fetch API on CF's worker didn't support the `credentials` field. So the SDK from svix cause the issue by set it to `same-origin`.

https://github.com/svix/svix-webhooks/blob/e5dbc7556f699e5f62092d2f57e979333f775534/javascript/src/openapi/http/isomorphic-fetch.ts#L40

The fix can be easy like the last comment on that issue:
https://github.com/cloudflare/workers-sdk/issues/2514#issuecomment-2178070014

**Comment by BernardoSM** [comment]
> The Fetch API on CF's worker didn't support the `credentials` field. So the SDK from svix cause the issue by set it to `same-origin`.
> 
> https://github.com/svix/svix-webhooks/blob/e5dbc7556f699e5f62092d2f57e979333f775534/javascript/src/openapi/http/isomorphic-fetch.ts#L40
> 
> The fix can be easy like the last comment on that issue: [cloudflare/workers-sdk#2514 (comment)](https://github.com/cloudflare/workers-sdk/issues/2514#issuecomment-2178070014)

I don't have access to fetch functions because I'm using svix SDK. Right now I removed svix sdk and add my own fetch functions to access svix api.

I was wondering if it makes sense to add this fix you mentioned inside svix sdk and not on our (svix customer) side. CloudFlare workers are getting more and more famous nowadays and it would increase the developer experience if I could start a CF worker and add svix with no errors.

**Comment by tasn** [comment]
Yeah, I think it's probably a good idea to do it. Though it's annoying that this needs to be done in the first place.
